src/cpp/session/SessionSourceDatabaseSupervisor.cpp

```diff
@@ -354,18 +351,29 @@ bool reclaimOrphanedSession(FilePath* pSessionDir)
 
    BOOST_FOREACH(const FilePath& sessionDir, sessionDirs)
    {
+      // if the suspend file exists, this session is only sleeping, not dead
+      if (sessionSuspendFilePath(sessionDir).exists())
```

Comment:
Ah, right -- sorry, I was thinking specifically in the context of a restart (which is an explicit suspend + resume action); ie, thinking specifically if we can / should attempt to detect a failed restart.